libcloudforensics/logging_utils.py

```diff
@@ -0,0 +1,120 @@
+
+
+def _GenerateColorSequences() -> List[str]:
+  """Generates ANSI codes for 256 colors.
+  Works on Linux and macOS, Windows (WSL) to be confirmed.
+
+  Returns:
+    List[str]: A list of ANSI codes.
+  """
+  sequences = []
+  for i in range(0, 16):
+    for j in range(0, 16):
+      code = str(i * 16 + j)
+      seq = '\u001b[38;5;' + code + 'm'
+      sequences.append(seq)
+  return sequences
+
+
+COLOR_SEQS = _GenerateColorSequences()
+RESET_SEQ = '\u001b[0m'
+
+# Cherrypick a few interesting values. We still want the whole list of colors
+# so that modules have a good amount colors to chose from.
+# pylint: disable=unbalanced-tuple-unpacking
+BLACK, RED, GREEN, YELLOW, BLUE, MAGENTA, CYAN, WHITE = COLOR_SEQS[8:16]
+BG_RED = '\u001b[41m'  # Red background
+BOLD = '\u001b[1m'  # Bold / bright modifier
+
+# We'll get something like this:
+# [2020-07-09 18:06:05,187] [libcloudforensics] INFO   Disk successfully copied
+LOG_FORMAT = ('[%(asctime)s] [{0:s}{color:s}%(name)-20s{1:s}] %(levelname)-8s'
+              ' %(message)s')
+
+LEVEL_COLOR_MAP = {
+    'WARNING': YELLOW,
+    'INFO': WHITE,
+    'DEBUG': BLUE,
+    'CRITICAL': BOLD + BG_RED + WHITE,
+    'ERROR': RED
+}
+
+
+class Formatter(logging.Formatter):
+  """Helper class used to add color to log messages depending on their level."""
+
+  def __init__(self,
+               colorize: bool = True,
+               random_color: bool = False,
+               **kwargs: str) -> None:
+    """Initializes the Formatter object.
+
+    Args:
+      colorize (bool): If True, output will be colorized.
+      random_color (bool): If True, will colorize the module name with a random
+          color picked from COLOR_SEQS.
+    """
+    self.colorize = colorize
+    kwargs['fmt'] = LOG_FORMAT.format('', '', color='')
+    if self.colorize:
+      color = ''
+      if random_color:
+        color = random.choice(COLOR_SEQS)
+      kwargs['fmt'] = LOG_FORMAT.format(BOLD, RESET_SEQ, color=color)
+    super(Formatter, self).__init__(**kwargs)
+
+  def format(self, record: logging.LogRecord) -> str:
+    """Hooks the native format method and colorizes messages if needed.
+
+    Args:
+      record (logging.LogRecord): Native log record.
+
+    Returns:
+      str: The formatted message string.
+    """
+    if self.colorize:
+      message = record.getMessage()
+      loglevel_color = LEVEL_COLOR_MAP.get(record.levelname)
+      if loglevel_color:
+        message = loglevel_color + message + RESET_SEQ
+      record.msg = message
+    return super(Formatter, self).format(record)
+
+
+def GetLogger(name: str) -> logging.Logger:
+  """Return a logger.
+
+  Args:
+    name (str): The name for the logger.
+
+  Returns:
+    logging.Logger: A logger.
+  """
+  logger = logging.getLogger(name)
+  logger.setLevel(logging.INFO)
+  console_handler = logging.StreamHandler(sys.stdout)
+  formatter = Formatter(random_color=True)
+  console_handler.setFormatter(formatter)
+  logger.addHandler(console_handler)
```

Comment:
This will add a new console_handler every time you call `GetLogger`. For each handler added, a log line will be displayed when you call the logging functions... so you might want to rename that function to SetUp logger or something, and use `logging.getLogger(name)` directly in other modules that need to log things.

Reply:
but you need to add a console handler for each new logger being created no? E.g. if calling two times getLogger but with different names, don't you need to both times create and add the console handler? In the meantime, I added a safeguard so that if the same logger is queried, no new handler is created.

Reply:
Correct. But if for some reason `GetLogger` is called twice with the same name, since loggers are singletons, handlers will be added twice. The check you added should be enough to not have this problem; but I'd still rename the method to `SetupLogger` since you're doing more than just fetching the singleton.

Reply:
Added